src/linear_algebra/affine_space/affine_equiv.lean

```diff
@@ -162,6 +165,24 @@ def mk' (e : P₁ → P₂) (e' : V₁ ≃ₗ[k] V₂) (p : P₁) (h : ∀ p' :
 
 @[simp] lemma symm_linear (e : P₁ ≃ᵃ[k] P₂) : e.linear.symm = e.symm.linear := rfl
 
+-- `simps` is not happy with the order we have the variables in this section, so we attach them all
+-- to these lemmas manually.
+
+/-- See Note [custom simps projection] -/
+def simps.apply {k P₁ P₂ V₁ V₂ : Type*} [ring k]
+  [add_comm_group V₁] [module k V₁] [add_torsor V₁ P₁]
+  [add_comm_group V₂] [module k V₂] [add_torsor V₂ P₂]
+  (e : P₁ ≃ᵃ[k] P₂) : P₁ → P₂ := e
+
+/-- See Note [custom simps projection] -/
+def simps.symm_apply {k P₁ P₂ V₁ V₂ : Type*} [ring k]
+  [add_comm_group V₁] [module k V₁] [add_torsor V₁ P₁]
+  [add_comm_group V₂] [module k V₂] [add_torsor V₂ P₂] (e : P₁ ≃ᵃ[k] P₂) :
+  P₂ → P₁ := e.symm
```

Comment:
@fpvandoorn, any idea why this is happening?

Reply:
You state it yourself:

> `simps` is not happy with the order we have the variables in this section

The order of the variables is different for the variables in the section than it is in the definition of `affine_equiv`.
(`affine_equiv (k P₁ P₂ : Type*) {V₁ V₂ : Type*}` vs `variables {k V₁ V₂ V₃ V₄ P₁ P₂}`).
This means that `simps.symm_apply` has the wrong type, so it is rejected.

The easy fix is to change the order of the (implicit) variables in this file.

Reply:
So this is working as intended then, simps has no intention of being able to reorder binders?

Reply:
correct